common/dynamicconfig/configstore/config_store_client.go

```diff
@@ -342,6 +351,19 @@ func (csc *configStoreClient) ListValue(name dc.Key) ([]*types.DynamicConfigEntr
 	return resList, nil
 }
 
+func (csc *configStoreClient) Stop() {
+	if !atomic.CompareAndSwapInt32(&csc.status, common.DaemonStatusStarted, common.DaemonStatusStopped) {
+		return
+	}
+	close(csc.doneCh)
+}
+
+func (csc *configStoreClient) Start() {
```

Comment:
where is it being called?

Reply:
it's not being called

Reply:
nobody is calling start and stop, and the goroutine of updating cache is leaked

Reply:
I should have provided better context. I'm pulling the store out into it's own field in resource and stop is being called on resource stop, similar to the isolation-group.

Reply:
you should make it in the same diff

Reply:
I'll leave it in the larger refactoring diff then